aptos-move/framework/aptos-framework/sources/staking_contract.move

```diff
@@ -150,37 +150,65 @@ module aptos_framework::staking_contract {
         amount: u64,
     }
 
+    #[view]
+    /// Return the address of the underlying stake pool for the staking contract between the provided staker and
+    /// operator.
+    ///
+    /// This errors out the staking contract with the provided staker and operator doesn't exist.
     public fun stake_pool_address(staker: address, operator: address): address acquires Store {
         assert_staking_contract_exists(staker, operator);
         let staking_contracts = &borrow_global<Store>(staker).staking_contracts;
         simple_map::borrow(staking_contracts, &operator).pool_address
     }
 
+    #[view]
+    /// Return the last recorded principal (the amount that 100% belongs to the staker with commission already paid for)
+    /// for staking contract between the provided staker and operator.
+    ///
+    /// This errors out the staking contract with the provided staker and operator doesn't exist.
     public fun last_recorded_principal(staker: address, operator: address): u64 acquires Store {
         assert_staking_contract_exists(staker, operator);
         let staking_contracts = &borrow_global<Store>(staker).staking_contracts;
         simple_map::borrow(staking_contracts, &operator).principal
     }
 
+    #[view]
+    /// Return percentage of accumulated rewards that will be paid to the operator as commission for staking contract
+    /// between the provided staker and operator.
+    ///
+    /// This errors out the staking contract with the provided staker and operator doesn't exist.
     public fun commission_percentage(staker: address, operator: address): u64 acquires Store {
         assert_staking_contract_exists(staker, operator);
         let staking_contracts = &borrow_global<Store>(staker).staking_contracts;
         simple_map::borrow(staking_contracts, &operator).commission_percentage
     }
 
+    #[view]
+    /// Return a tuple of three numbers:
+    /// 1. The total active stake in the underlying stake pool
+    /// 2. The total accumulated rewards that haven't had commission paid out
```

Comment:
Could you clarify this? Perhaps an additional comment like:

> Therefore, the total accumulated rewards for the shareholder would be `<2> - <3>`.

Assuming that my understanding is actually correct.

Reply:
Note that this is the staking_contract, not vesting. To the staking_contract, the entire vesting pool is the staker. (2) - (3) would only be the net new accumulated rewards (not yet tracked in principal). So current principal + (2) - (3) would be the new principal, or what 100% belong to the staker (with all due commission already paid).

I'd not rather not explain that here as that's way too complex.